type-c-service/src/wrapper/pd.rs

```diff
@@ -31,8 +42,14 @@ impl<const N: usize, C: Controller> ControllerWrapper<'_, N, C> {
     async fn process_controller_command(
         &self,
         controller: &mut C,
+        state: &mut InternalState,
         command: &controller::InternalCommandData,
     ) -> Response<'static> {
+        if state.fw_update_state.in_progress() {
+            debug!("FW update in progress, ignoring controller command");
+            return controller::Response::Controller(Err(PdError::Busy));
+        }
```

Comment:
Could this check go in `process_pd_command()` or even the top-level `process()`?